File: Python3/566. Reshape the Matrix.py
class Solution:
    def matrixReshape(self, nums, r, c):
        """
        :type nums: List[List[int]]
        :type r: int
        :type c: int
        :rtype: List[List[int]]
        """
        num1D = [x for row in nums for x in row]
        if len(nums)*len(nums[0]) != r*c or len(nums) == r:
            return nums
        else:
            return [num1D[i*c:i*c+c] for i in range(r)]

File: Golang/566. Reshape the Matrix.go
func matrixReshape(nums [][]int, r int, c int) [][]int {
    n := len(nums) // origin row number
    m := len(nums[0]) // origin col number
    if n*m != r*c || n == r {
        return nums
    }
    
    count := 0
    matrix := make([][]int, r)
    
    for i := 0; i < r; i++ {
        matrix[i] = make([]int, c)
        for j := 0; j < c; j++ {
            matrix[i][j] = nums[count/m][count%m]
            count++
        }
    }
    
    return matrix
}